actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -474,6 +474,247 @@ def apply_form_for_options!(record, object, options) #:nodoc:
       end
       private :apply_form_for_options!
 
+      # Creates a form tag based on mixing URLs, scopes, or models.
+      #
+      #   # Using just a URL:
+      #   form_with url: posts_path do |form|
+      #     form.text_field :title
+      #   end
+      #   # =>
+      #   <form action="/posts" method="post" data-remote="true">
+      #     <input type="text" name="title">
+      #   </form>
+      #
+      #   # Adding a scope prefixes the input field names:
+      #   form_with scope: :post, url: posts_path do |form|
+      #     form.text_field :title
+      #   end
+      #   # =>
+      #   <form action="/posts" method="post" data-remote="true">
+      #     <input type="text" name="post[title]">
+      #   </form>
+      #
+      #   # Using a model infers both the URL and scope:
+      #   form_with model: Post.new do |form|
+      #     form.text_field :title
+      #   end
+      #   # =>
+      #   <form action="/posts" method="post" data-remote="true">
+      #     <input type="text" name="post[title]">
+      #   </form>
+      #
+      #   # An existing model makes an update form and fills out field values:
+      #   form_with model: Post.first do |form|
+      #     form.text_field :title
+      #   end
+      #   # =>
+      #   <form action="/posts/1" method="post" data-remote="true">
+      #     <input type="hidden" name="_method" value="patch">
+      #     <input type="text" name="post[title]" value="<the title of the post>">
+      #   </form>
+      #
+      # The parameters in the forms are accessible in controlleres according to
+      # their name nesting. So inputs named +title+ and <tt>post[title]</tt> are
+      # accessible as <tt>params[:title]</tt> and <tt>params[:post][:title]</tt>
+      # respectively.
+      #
+      # By default +form_with+ attaches the <tt>data-remote</tt> attribute
+      # submitting the form via an XMLHTTPRequest in the background if an
+      # an Unobtrusive JavaScript driver, like jquery-ujs, is used. See the
+      # <tt>:remote</tt> option for more.
+      #
+      # For ease of comparison the examples above left out the submit button,
+      # as well as the auto generated hidden fields that enable UTF-8 support
+      # and adds an authenticity token needed for Cross Site Request Forgery
```

Comment:
Don't think this needs title case. It is referenced as small case in its own documentation.